### PR TITLE
feat(security): expose the safe value types.

### LINKED ISSUE #8568: Include the Safe* interfaces in the exports of @angular/platform-browser
After reading #8491, I tried to implement the solution to [mark the generated style string as safe](https://github.com/angular/angular/issues/8491#issuecomment-217467582). 
```ts
this.svgUrl = sanitizer.bypassSecurityTrustStyle('url(\'image.svg\')');
```
However, `bypassSecurityTrustStyle()` returns `SafeStyle` instead of `string` (as the comment suggests) and this marker interface isn't exported by `@angular/platform-browser`. That means, I cannot properly type my class member.

Please export the marker interfaces `SafeStyle`, `SafeHtml`, etc. so this solution works without `any`.

### PR DESCRIPTION
**Pull request by mprobst**
This allows users to properly type their `SafeHtml`, `SafeStyle`, etc values.

Fixes #8568.

### CODE REVIEW

**Comment by mprobst** [comment]
@rjamet @mhevery

**Comment by rjamet** [comment]
LGTM. 

If for some reason you see patterns of people passing things that are semantically templates or template URLs, it might be interesting to make a separate type (template could require a constant string, for instance). We're still debating the usefulness of this internally.

File: modules/@angular/integration_test/public_api_spec.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ import {ListWrapper} from '../facade/src/collection';
 //
 // =================================================================================================
 // =================================================================================================
+// Note that this test only tests for *values* exported (e.g. classes, functions, variables), but
+// not for *types* exported (interfaces, typedefs).
+// =================================================================================================
+
```
Comment (mhevery) [inline modules/@angular/integration_test/public_api_spec.ts]: we have a separate test for types which is currently broken. @IgorMinar is working on getting it working in the new world.

Comment (mprobst) [inline modules/@angular/integration_test/public_api_spec.ts]: Ack. I'll leave the comment here, and point to the other test – note that I did modify the code there in this PR, it just happens to not have any effect for the time being...

**Pull request closed without merging**